pkg/cloud/ratelimit.go

```diff
@@ -112,3 +112,108 @@ func (m *MinimumRateLimiter) Accept(ctx context.Context, key *RateLimitKey) erro
 func (m *MinimumRateLimiter) Observe(ctx context.Context, err error, key *RateLimitKey) {
 	m.RateLimiter.Observe(ctx, err, key)
 }
+
+type TickerRateLimiter struct {
+	ticker *time.Ticker
+}
+
+func NewTickerRateLimiter(limit int, interval time.Duration) *TickerRateLimiter {
+	return &TickerRateLimiter{
+		ticker: time.NewTicker(interval / time.Duration(limit)),
+	}
+}
+
+// Accept everything immediately.
+func (t *TickerRateLimiter) Accept(ctx context.Context, rlk *RateLimitKey) error {
+	select {
+	case <-t.ticker.C:
+		return nil
+	case <-ctx.Done():
+		return ctx.Err()
+	}
+}
+
+// Observe does nothing.
+func (*TickerRateLimiter) Observe(context.Context, error, *RateLimitKey) {
+}
+
+// CompositeRateLimiter combines rate limiters based on RateLimitKey.
+type CompositeRateLimiter struct {
+	rateLimiters map[string]map[string]map[string]RateLimiter
```

Comment:
please add in comment what those string stands for. I understand that the order is project, service and operation? It doesn't align with RateLimitKey variables order and is not clear when looking at this.

Reply:
why `resource name` when you call this service in code. I think service is a good choice